packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -390,18 +409,20 @@ function shallowCompatibilityTest(
 		return data[typeNameSymbol] === schema.identifier;
 	}
 
+	// Array schemas strictly allow array data as input
```

Comment:
If you look at SchemaFactory.array, you can see the allowed insertable data type is `Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>`, so not just arrays.

Instead I suggest checking for Symbol.iterator, and allowing that for both maps and arrays.

Reply:
For now, I've reverted the functional changes to this code. We can do a follow-up PR to change the input contracts.